src/Serilog.Sinks.Loggly/Sinks/Loggly/FileBufferDataProvider.cs

```diff
@@ -0,0 +1,279 @@
+                {
+                    // Count is the indicator that work was done, so advances even in the (rare) case an
+                    // oversized event is dropped.
+                    ++count;
+
+                    if (_eventBodyLimitBytes.HasValue 
+                        && nextLine != null
+                        && _encoding.GetByteCount(nextLine) > _eventBodyLimitBytes.Value)
+                    {
+                        SelfLog.WriteLine(
+                            "Event JSON representation exceeds the byte size limit of {0} and will be dropped; data: {1}",
+                            _eventBodyLimitBytes, nextLine);
+                    }
+                    if (!nextLine.StartsWith("{"))
+                    {
+                        //in some instances this can happen. TryReadLine no longer assumes a BOM if reading from the file start, 
+                        //but there may be (unobserved yet) situations where the line read is still not complete and valid 
+                        // Json. This and the try catch that follows are, therefore, attempts to preserve the 
+                        //logging functionality active, though some events may be dropped in the process.
+                        SelfLog.WriteLine(
+                            "Event JSON representation does not start with the expected '{{' character. " +
+                            "This may be related to a BOM issue in the buffer file. Event will be dropped; data: {0}",
+                            nextLine);
+                    }
+                    else
+                    {
+                        try
+                        {
+                            events.Add(DeserializeEvent(nextLine));
+                        }
+                        catch (Exception ex)
+                        {
+                            SelfLog.WriteLine(
+                                "Unable to deserialize the json event; Event will be dropped; exception: {0};  data: {1}",
+                                ex.Message, nextLine);
+                        }
+                    }
+                }
+            }
+
+            _futureBookmark = new Bookmark(positionTracker, _currentBookmark.FileName);
+            return events;
+        }
+
+        // It would be ideal to chomp whitespace here, but not required.
+        bool TryReadLine(Stream current, ref long nextStart, out string nextLine)
+        {
+            // determine if we are reading the first line in the file. This will help with 
+            // solving the BOM marker issue ahead
+            var firstline = nextStart == 0;
+
+            if (current.Length <= nextStart)
+            {
+                nextLine = null;
+                return false;
+            }
+
+            current.Position = nextStart;
+
+            // Important not to dispose this StreamReader as the stream must remain open.
+            var reader = new StreamReader(current, _encoding, false, 128);
```

Comment:
you can use the ctor that has the _leaveOpen_ flas to make it explicit that the underlying stream will remain open
https://msdn.microsoft.com/en-us/library/gg712952(v=vs.110).aspx